Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -541,13 +560,46 @@ namespace ROS2::Utils
         scene->GetManifest().SaveToFile(assetInfoFile.Native());
         AZ_Printf("CreateSceneManifest", "Saving scene manifest to %s\n", assetInfoFile.c_str());
 
+        assetsFilenames = GetMeshAssets(scene);
+        AZ_Printf("CreateSceneManifest", "Found %d assets for %s scene\n", assetsFilenames.size(), assetInfoFile.c_str());
+
         return true;
     }
 
-    bool CreateSceneManifest(const AZ::IO::Path& sourceAssetPath, bool collider, bool visual)
+    bool CreateSceneManifest(
+        const AZ::IO::Path& sourceAssetPath, const bool collider, const bool visual, AZStd::unordered_set<AZStd::string>& assetsFilenames)
+    {
+        return CreateSceneManifest(sourceAssetPath, sourceAssetPath.Native() + ".assetinfo", collider, visual, assetsFilenames);
+    }
+
+    AZStd::unordered_set<AZStd::string> GetMeshAssets(const AZStd::shared_ptr<AZ::SceneAPI::Containers::Scene>& scene)
     {
-        auto assetInfoFilePath = AZ::IO::Path{ sourceAssetPath };
-        assetInfoFilePath.Native() += ".assetinfo";
-        return CreateSceneManifest(sourceAssetPath, sourceAssetPath.Native() + ".assetinfo", collider, visual);
+        AZStd::unordered_set<AZStd::string> assetsFilenames;
+
+        // Look for material files
+        AZStd::initializer_list<AZ::SceneAPI::DataTypes::IMaterialData::TextureMapType> allTextureTypes = {
```

Comment:
consider const or even better constexpr.

Reply:
I am not sure if `constexpr` is possible in this configuration. Added `static const`.